### ISSUE TITLE
Bliss shaders crash in the Nether and End

### ISSUE THREAD

**Issue by 9182736450typist** [X0nk/Bliss-Shader#365]
Computer Specs:
Macbook Pro 14" with M4 pro, 24GB RAM and 1TB SSD
OS: MacOS Sequoia 15.1
Issue: Bliss shaders crashes in the Nether or the End. I tried both entering the nether/end and loading the shader while IN the nether/end. In all instances the game quits and crashes.

Crash report:

A fatal error has been detected by the Java Runtime Environment:
#
#  SIGSEGV (0xb) at pc=0x00000001ea83e02c, pid=25864, tid=259
#
# JRE version: OpenJDK Runtime Environment Microsoft-8035246 (17.0.8+7) (build 17.0.8+7-LTS)
# Java VM: OpenJDK 64-Bit Server VM Microsoft-8035246 (17.0.8+7-LTS, mixed mode, tiered, compressed oops, compressed class ptrs, g1 gc, bsd-aarch64)
# Problematic frame:
# C  [libGLProgrammability.dylib+0x8102c]  glpLLVMGetFunctionGlobalVariableUse+0x74
#
# No core dump will be written. Core dumps have been disabled. To enable core dumping, try "ulimit -c unlimited" before starting Java again
#
# If you would like to submit a bug report, please visit:
#   https://aka.ms/minecraftjavacrashes
# The crash happened outside the Java Virtual Machine in native code.
# See problematic frame for where to report the bug.
#

---------------  S U M M A R Y ------------

Command Line: -Djava.library.path=/Library/Application Support/minecraft/bin/7deadf2e21613fe3d7822104ddb7e94014f6aaa1 -Djna.tmpdir=/Library/Application Support/minecraft/bin/7deadf2e21613fe3d7822104ddb7e94014f6aaa1 -Dorg.lwjgl.system.SharedLibraryExtractPath=/Library/Application Support/minecraft/bin/7deadf2e21613fe3d7822104ddb7e94014f6aaa1 -Dio.netty.native.workdir=/Library/Application Support/minecraft/bin/7deadf2e21613fe3d7822104ddb7e94014f6aaa1 -Dminecraft.launcher.brand=minecraft-launcher -Dminecraft.launcher.version=3.5.17 -DFabricMcEmu= net.minecraft.client.main.Main  -Xmx9G -XX:+UnlockExperimentalVMOptions -XX:+UseG1GC -XX:G1NewSizePercent=20 -XX:G1ReservePercent=20 -XX:MaxGCPauseMillis=50 -XX:G1HeapRegionSize=32M -Dorg.lwjgl.opengl.disable=false -Dsun.java2d.opengl=true -Dlog4j.configurationFile=/Library/Application Support/minecraft/assets/log_configs/client-1.12.xml net.fabricmc.loader.impl.launch.knot.KnotClient --username MrPromethium --version fabric-loader-0.15.11-1.20.1 --gameDir /Documents/curseforge/minecraft/Instances/Fabulously Optimized Exp --assetsDir /Library/Application Support/minecraft/assets --assetIndex 5 --uuid b94086bc90024c368fc7f217b78c34e4 -IIFtkIfzdrdWvhqDweSGc-h5aLZVSPNj52jNnlCU --clientId RTMzRkEwNDM3NzkzNENFMkEyRTU3N0JCMEREQkVGRTE= --xuid 2535419467916809 --userType msa --versionType release --quickPlayPath /Library/Application Support/minecraft/quickPlay/java/1733696252018.json

Host: "Mac16,8" arm64, 14 cores, 24G, Darwin 24.1.0, macOS 15.1 (24B2082)
Time: Sun Dec  8 14:21:28 2024 PST elapsed time: 236.362114 seconds (0d 0h 3m 56s)

---------------  T H R E A D  ---------------

Current thread (0x0000000103808a00):  JavaThread "Render thread" [_thread_in_native, id=259, stack(0x000000016c640000,0x000000016ce3c000)]

Stack: [0x000000016c640000,0x000000016ce3c000],  sp=0x000000016ce33820,  free space=8142k
Native frames: (J=compiled Java code, j=interpreted, Vv=VM code, C=native code)
C  [libGLProgrammability.dylib+0x8102c]  glpLLVMGetFunctionGlobalVariableUse+0x74
C  [libGLProgrammability.dylib+0x81244]  glpLLVMGetFunctionGlobalVariableUse+0x28c
C  [libGLProgrammability.dylib+0x81244]  glpLLVMGetFunctionGlobalVariableUse+0x28c
C  [libGLProgrammability.dylib+0x81244]  glpLLVMGetFunctionGlobalVariableUse+0x28c
C  [libGLProgrammability.dylib+0x81244]  glpLLVMGetFunctionGlobalVariableUse+0x28c
C  [libGLProgrammability.dylib+0x7fa9c]  glpLLVMCGFindSamplersAndBuffers+0x48c
C  [libGLProgrammability.dylib+0x7e928]  glpLLVMCGTopLevel+0x570
C  [libGLProgrammability.dylib+0xa5aa8]  glpLinkProgram+0x2718
C  [libGLProgrammability.dylib+0xbf81c]  ShLink+0xd0
C  [GLEngine+0xfe410]  gleLinkProgram+0x444
C  [GLEngine+0x783f8]  glLinkProgramARB_Exec+0xb4
C  0x7851000139d7b63c
J 44313 c1 com.mojang.blaze3d.platform.GlStateManager.glLinkProgram(I)V (8 bytes) @ 0x0000000132d3af7c [0x0000000132d3adc0+0x00000000000001bc]
C  0x000000015c5c9ba8

**Comment by X0nk** [comment]
i cannot determine any useful information from this crash report unfortunately. i need more relevant information to do anything.
it would be better to get the `latest.log` or output log up to when the game closes, it logs whats going on as it happens in the game, and thats where shader debug information goes as well.

on top of that, i have no clue what your mod loader setup is, a random mod could also be the cause of it entirely.
from the information youve given, i dont even know if it *is* the shader causing it.

since youre on apple, ill link this just in case lol [issue #181](https://github.com/X0nk/Bliss-Shader/issues/181)

**Comment by 9182736450typist** [comment]
Update: Bliss 2.0.3 Nether works fine. The End still crashes.

**Comment by 9182736450typist** [comment]
FIXED! Unzipped the file and deleted the world1 folder, so it doesn't crash in the end!

**Comment by X0nk** [comment]
my friend thats not a fix. if you would please provide more information, i could *actually* fix it

**Comment by X0nk** [comment]
could you perhaps test this version of the shader? [this will download the in-development version on click.](https://github.com/X0nk/Bliss-Shader/archive/refs/heads/main.zip)

**Comment by w1lfpi** [comment]
> ИСПРАВЛЕНО! Распаковал файл и удалил папку world1, чтобы в конце концов она не сбоила!

Hi, I also can’t enter the Nether and the End. Can you explain in more detail what you did to fix this issue

> FIXED! Unzipped the file and deleted the world1 folder, so it doesn't crash in the end!

Hi, I also can’t enter the Nether and the End. Can you explain in more detail what you did to fix this issue

**Comment by w1lfpi** [comment]
Hi, I also can’t enter the Nether and the End. Can you explain in more detail what you did to fix this issue

> FIXED! Unzipped the file and deleted the world1 folder, so it doesn't crash in the end!

**Comment by w1lfpi** [comment]
> could you perhaps test this version of the shader? [this will download the in-development version on click.](https://github.com/X0nk/Bliss-Shader/archive/4ba7728cdc803a781b3603d75556e69f0ecfaca3.zip)

This version also crashes End and Add.

**Comment by X0nk** [comment]
could you provide information??? like a screenshot of your f3 debug overlay or something?

**Comment by w1lfpi** [comment]
> не могли бы вы предоставить информацию??? как скриншот вашего наложения отладки f3 или что-то в этом роде?

[Uploading hs_err_pid8958.log…]()

**Comment by X0nk** [comment]
closing this issue because... yeah... i have nothing to go on, and its gone stale

**Comment by MammaGrus** [comment]
This issue is still present on the latest version v2.0.4. Here is my latest log file and crash log: I'm on the latest version of MacOS and playing on Minecraft 1.21.4. It been an issue since last year. Please ask if you need any more information. I really wish this to be resolved. 

[latest.log](https://github.com/user-attachments/files/20081178/latest.log)

[hs_err_pid26751.log](https://github.com/user-attachments/files/20081204/hs_err_pid26751.log)

**Comment by MammaGrus** [comment]
Should I start a new issue or is commenting here fine?

**Comment by X0nk** [comment]
did you enable the POM/parallax occlusion mapping setting? if i remember right, that doesnt work on apple

**Comment by MammaGrus** [comment]
It is not turned on as that just doesn't work at all

**Comment by X0nk** [comment]
well, a latest.log after the error happens is about all you can give me, or if youre using iris, the blue "copy info" text, paste it here

**Comment by MammaGrus** [comment]
I am using Iris but the game crashes immediately. Therefore, I am unable to provide the blue text error message.

This is the latest.log file 
[hs_err_pid92788.log](https://github.com/user-attachments/files/20351602/hs_err_pid92788.log)

**Comment by MammaGrus** [comment]
[latest.log](https://github.com/user-attachments/files/20351617/latest.log) or this one perhaps

**Comment by X0nk** [comment]
i see distant horizons in your loaded mods, and that you are an apple user.
[please refer to this issue](https://github.com/X0nk/Bliss-Shader/issues/181)

**Comment by MammaGrus** [comment]
Sure, but that's not the issue. The game still crashes with DH turned off. Or do you mean to say that the crash in the nether is tied to the same problem?

**Comment by X0nk** [comment]
are your settings default/stock settings?

**Comment by MammaGrus** [comment]
Yes they are

**Comment by MammaGrus** [comment]
Did a fresh install of the shader to test

**Comment by Lium62063** [comment]
Same issue here. I did what OP said and deleted "world1" (not world-1) and it doesnt CRASH but it also makes the End look really weird, the sky does not look like how it is shown in youtube videos.

Then I did more than what the OP did and kept the "world1" folder but deleted individual files in the folder, i found that none of the "gbuffer" files are causing the issue, it's all the other files that cause the issue. BUT when i delete those, it doesnt crash although entering the end it's just pitch black except for my hotbar. 

Yes i am using mac. It is definitely not a RAM issue i allocated 12gb when 2gb is default. And bliss works perfectly fine in overworld. 

I never tried adjusting the nether files (i assume world-1 is nether) although im guessing it is the same problem

**Comment by X0nk** [comment]
deleting files will break the visuals, yes that's expected.
the gbuffer files *are* used, but they can fallback to other gbuffer programs if they aren't used, so it may just be doing that instead of breaking. i'm sure it breaks visually in some places, but that shouldn't cause a compile error or anything.

**Comment by X0nk** [comment]
okay... i want to make absolutely sure that everyone of us is on the same exact page. at this point
download and test [THIS specific bliss version](https://github.com/X0nk/Bliss-Shader/archive/refs/heads/unstable-development.zip)
with default settings, and no additional mods like distant horizons, physics mod.
and note your findings, please do not use a different version

**Comment by MammaGrus** [comment]
I tried in a new instance of minecraft 1.21.4 with only iris and sodium, and this version doesn't crash the game! I then tried it on my normal, cluttered and modded instance and lo and behold it works without crash here as well.

**Comment by Lium62063** [comment]
Well it does work! i tried using the link you provided. Nether and end both fine It doesn't work with distant horizons though. Kinda confused (and bummed) cuz i saw people on youtube using Bliss + distant horizons but oh well

**Comment by Lium62063** [comment]
Im kinda bummed what way did the youtubers use to use distant horizons + bliss cuz that looks cool

**Comment by X0nk** [comment]
they simply weren't using an apple device. it works without issue on AMD, Intel, Nvidia, its just apple. take a look at this issue #181 for more of an explanation.
